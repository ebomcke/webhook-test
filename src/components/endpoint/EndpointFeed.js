import React from 'react';
import { Feed, Button } from 'semantic-ui-react';
import EndpointItem from './EndpointItem';
import { compose } from 'recompose';
import { withNavigation } from '../../contexts/Navigation';
import { withEndpoints } from '../../contexts/Endpoints';

const EndpointFeed = ({ endpoints, navigate }) => (
  <Feed>
    <Button
      size="big"
      content="New endpoint"
      icon="plus"
      onClick={() =>
        navigate({
          name: 'NewEndpoint',
        })
      }
    />
    {endpoints.map(endpoint => (
      <EndpointItem key={endpoint.id} endpoint={endpoint} />
    ))}
  </Feed>
);

export default compose(withNavigation, withEndpoints)(EndpointFeed);
