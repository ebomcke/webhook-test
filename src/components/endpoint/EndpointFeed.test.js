import React from 'react';
import { create } from 'react-test-renderer';
import EndpointFeed from './EndpointFeed';
import { Button } from 'semantic-ui-react';
jest.mock('./EndpointItem');

describe('Rendering tests', () => {
  test('Basic', () => {
    const endpointFeed = create(<EndpointFeed />);
    expect(endpointFeed.toJSON()).toMatchSnapshot();
  });
});

describe('Interactions tests', () => {
  test('Click on new endpoints button navigates to new endpoint view', () => {
    const navigate = jest.fn();
    const endpointFeed = create(<EndpointFeed navigate={navigate} />);
    const button = endpointFeed.root.findByType(Button);
    button.props.onClick();
    expect(navigate.mock.calls.length).toBe(1);
    expect(navigate.mock.calls[0][0]).toEqual({
      name: 'NewEndpoint',
    });
  });
});
