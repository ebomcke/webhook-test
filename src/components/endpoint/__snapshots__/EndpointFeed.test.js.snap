// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering tests Basic 1`] = `
<div
  className="ui feed"
>
  <button
    className="ui big button"
    onClick={[Function]}
  >
    <i
      aria-hidden="true"
      className="plus icon"
    />
    New endpoint
  </button>
  <div>
    This is a placeholder for the endpoint item
    <pre>
      <code>
        {"endpoint":{"id":"endpoint-1","path":"unit-test-1","defaultResponse":{"statusCode":200,"body":"{\\"unitTest\\":true}"}}}
      </code>
    </pre>
  </div>
  <div>
    This is a placeholder for the endpoint item
    <pre>
      <code>
        {"endpoint":{"id":"endpoint-2","path":"unit-test-2","defaultResponse":{"statusCode":200,"body":"{\\"unitTest\\":true}"}}}
      </code>
    </pre>
  </div>
</div>
`;
